### PR TITLE
Add loggin via winston library

### PR DESCRIPTION
**Pull request by jaywink**
Default level info, can be changed via UVS_LOG_LEVEL.

### CODE REVIEW

File: src/verify.js [hunk 1]
```diff
@@ -7,7 +9,10 @@ const homeserverUrl = process.env.UVS_HOMESERVER_URL;
 function sanityCheckRequest(req, res, fields=[]) {
     if (!req.body) {
         res.status(400);
-        res.send('Invalid request: no JSON content found in body.');
+
+        const msg = 'Invalid request: no JSON content found in body.';
+        res.send(msg);
+        logger.log('warning', msg);
```
Comment (jaywink) [inline src/verify.js]: [winston] Unknown logger level: warning

File: src/routes.js [hunk 2]
```diff
@@ -3,14 +3,19 @@ const {
...
 const routes = {
     getHealth: (req, res) => {
+        requestLogger(req);
         res.send('👍');
     },
     postVerifyUser: async(req, res) => {
+        requestLogger(req);
         const checkResult = sanityCheckRequest(req, res, ['token']);
         if (!checkResult) {
+            logger.log('info', 'Request sanity check failed.');
```
Comment (Half-Shot) [inline src/routes.js]: Thought: Would be good to have requestIDs, like in Synapse so you can follow along a single request in Kibana.

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+function requestLogger(req) {
+    if (req.method === 'POST') {
+        logger.log('info', `${req.method} ${req.path}: ${stringify(req.body)}`);
+    } else {
+        logger.log('info', `${req.method} ${req.path}`);
+    }
+}
+
+function stringify(obj) {
```
Comment (Half-Shot) [inline src/utils.js]: Prefer moving this above it's caller, as it was confusing for a second.

File: src/utils.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    try {
+        return JSON.stringify(obj);
+    } catch (error) {
+        return obj;
```
Comment (Half-Shot) [inline src/utils.js]: This doesn't feel like a string?

**Review by Half-Shot — APPROVED** [review]
Looking generally good. As per usual, I've inserted my ideas as comments :)

File: src/routes.js (hunk 2 above)
Comment (jaywink) [reply src/routes.js]: Great idea, added!

Comment (jaywink) [reply src/routes.js]:
```
{
  requestId: '40e68649-4723-4401-b890-557d978ae309',
  level: 'info',
  message: 'POST /verify/user: {"token":"foobar"}',
  timestamp: '2020-09-15T09:17:39.730Z'
}
{
  requestId: '40e68649-4723-4401-b890-557d978ae309',
  level: 'debug',
  message: 'Request sanity check ok',
  timestamp: '2020-09-15T09:17:39.731Z'
}
{
  requestId: '40e68649-4723-4401-b890-557d978ae309',
  level: 'debug',
  message: 'Making request to: https://federator.modular.im/_matrix/federation/v1/openid/userinfo?access_token=redacted',
  timestamp: '2020-09-15T09:17:39.731Z'
}
{
  requestId: '40e68649-4723-4401-b890-557d978ae309',
  level: 'debug',
  message: 'Verify token failed: 401, {"date":"Tue, 15 Sep 2020 09:17:39 GMT","content-type":"application/json","content-length":"71","connection":"close","cache-control":"no-cache, no-store, must-revalidate","access-control-allow-origin":"*","access-control-allow-methods":"GET, HEAD, POST, PUT, DELETE, OPTIONS","access-control-allow-headers":"Origin, X-Requested-With, Content-Type, Accept, Authorization","strict-transport-security":"max-age=15724800; includeSubDomains"}, {"errcode":"M_UNKNOWN_TOKEN","error":"Access Token unknown or expired"}',
  timestamp: '2020-09-15T09:17:39.999Z'
}
{
  requestId: '40e68649-4723-4401-b890-557d978ae309',
  level: 'info',
  message: 'User token check failed.',
  timestamp: '2020-09-15T09:17:40.000Z'
}



```

Comment (jaywink) [reply src/routes.js]: Also now redacting the token from the POST logs.

Comment (Half-Shot) [reply src/routes.js]: ooo great :)

**Review by Half-Shot — APPROVED** [review]
happy happy

**Pull request merged**